### PR TITLE
disable cmd cr

### PR DESCRIPTION
**Pull request by amirbilu**
- add enable/disable cmds
- add TabnineToggle command
- cr
- cr

### CODE REVIEW

**Comment by amirbilu** [comment]
@aarondill this is based on your PR. can you please review?

**Comment by aarondill** [comment]
So, if I'm understanding correctly, this:
- Updates the readme/ help
- adds lualine status
- adds TabnineStatus
- adds persistent state with a `.disabled` file
- reorganized a few things
- *removes* the TabnineToggle command

It looks good to me, but I have to ask, why did you remove the toggle command (especially since it's still mentioned in the PR description)

**Comment by amirbilu** [comment]
I'm not a big fan of toggle command. But If you think this is super helpful - I'll re-add it

**Review by aarondill — APPROVED** [review]

**Comment by aarondill** [comment]
Personally, I would actually use that ( it allows me to go up in my history to run the same command :) ), but this is certainly useable without it

File: lua/tabnine/state.lua [hunk 1]
```diff
@@ -1,4 +1,6 @@
 local uv = vim.loop
+local fn = vim.fn
+local utils = require("tabnine.utils")
```
Comment (aarondill) [inline lua/tabnine/state.lua:3]: Both of these are unused

Comment (aarondill) [reply lua/tabnine/state.lua:3]:
```lua
local api = vim.api
local fn = vim.fn
```
These definitions in `lua/tabnine/keymaps.lua` line 2-3 are also unused, though they were added / made unused in a separate PR (70281f8c2e4b8f91a1c2f66f2a716959f026cd9b)

**Review by aarondill — CHANGES_REQUESTED** [review]

File: lua/tabnine/state.lua (hunk 1 above)
Comment (aarondill) [inline lua/tabnine/state.lua:3]:
```suggestion
-local fn = vim.fn
-local utils = require("tabnine.utils")
```
both of these are unused.

edit: Created new comment to include GitHub suggestion

Comment (aarondill) [reply lua/tabnine/state.lua:3]:
```lua
local api = vim.api
local fn = vim.fn
```
These definitions in `lua/tabnine/keymaps.lua` line 2-3 are also unused, though they were added / made unused in a separate PR (70281f8c2e4b8f91a1c2f66f2a716959f026cd9b)

**Review by aarondill — CHANGES_REQUESTED** [review]

**Review by aarondill — APPROVED** [review]
Unless you'd like to re-add the `:TabnineToggle` command, then it looks good to me

**Comment by amirbilu** [comment]
I've applied your comments + added toggle command + move some logic to `status.lua`

**Comment by amirbilu** [comment]
Merging this for now. Highly appreciate your contribution! I know many people have been waiting for this!

File: lua/tabnine/status.lua [hunk 2]
```diff
@@ -0,0 +1,67 @@
+local uv = vim.loop
+local fn = vim.fn
+local utils = require("tabnine.utils")
+
+local M = {}
+local DISABLED_FILE = utils.script_path() .. "/.disabled"
+local tabnine_binary = require("tabnine.binary")
+local state = require("tabnine.state")
+local service_level = nil
+local status_prefix = "⌬ tabnine"
```
Comment (aarondill) [inline lua/tabnine/status.lua:10]: Should this be capital?

**Pull request merged**